### PR TITLE
[Validator] Fix `@Valid` regarding property `traverse` 

### PR DESCRIPTION
**Pull request by corphi**
While researching for symfony/symfony#27090, the constraint reference turned out to be incorrect.

### CODE REVIEW

File: reference/constraints/Valid.rst [hunk 1]
```diff
@@ -261,8 +261,8 @@ traverse
 
 **type**: ``boolean`` **default**: ``true``
 
-If this constraint is applied to a property that holds an array of objects,
-then each object in that array will be validated only if this option is
-set to ``true``.
+If this constraint is applied to a ``Traversable``, then all containing values
+will be validated if this option is set to ``true``. This option is ignored on
+arrays: Arrays are traversed in either case. Keys are not validated.
```
Comment (HeahDude) [inline reference/constraints/Valid.rst]: This should need a correction if https://github.com/symfony/symfony/pull/29800 is merged, right?

Comment (corphi) [reply reference/constraints/Valid.rst]: No, `@Valid` is working correctly. The docs could just use this clarification. The mentioned pull request is only concerned with other constraints not working correctly.

**Review by HeahDude — APPROVED** [review]
Thanks!

**Comment by javiereguiluz** [comment]
Thank you @corphi.

**Pull request merged**